src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -3112,7 +3119,8 @@ public static ref readonly TensorSpan<T> AtanPi<T>(scoped in ReadOnlyTensorSpan<
         public static T Average<T>(scoped in ReadOnlyTensorSpan<T> x)
             where T : INumberBase<T>
         {
-            return TensorPrimitivesHelperSpanInTOut(x, TensorPrimitives.Average);
+            T sum = Sum(x);
+            return sum / T.CreateChecked(x.FlattenedLength);
```

Comment:
nit: I wonder if we can do
```suggestion
            // Get the flattenedLength first so we don't spend time computing if we'll fail due to overflow
            T flattenedLength = T.CreateChecked(x.FlattenedLength);

            T sum = Sum(x);
            return sum / flattenedLength;
```